linera-views/src/map_view.rs

```diff
@@ -713,22 +710,26 @@ where
     /// # let context = create_memory_context();
     ///   let mut map = ByteMapView::load(context).await.unwrap();
     ///   map.insert(vec![0,1], String::from("Hello"));
-    ///   assert_eq!(map.get_mut_or_default(vec![7]).await.unwrap(), "");
-    ///   let value = map.get_mut_or_default(vec![0,1]).await.unwrap();
+    ///   assert_eq!(map.get_mut_or_default(&[7]).await.unwrap(), "");
+    ///   let value = map.get_mut_or_default(&[0,1]).await.unwrap();
     ///   assert_eq!(*value, String::from("Hello"));
     ///   *value = String::from("Hola");
     ///   assert_eq!(map.get(&[0,1]).await.unwrap(), Some(String::from("Hola")));
     /// # })
     /// ```
-    pub async fn get_mut_or_default(&mut self, short_key: Vec<u8>) -> Result<&mut V, ViewError> {
+    pub async fn get_mut_or_default(&mut self, short_key: &[u8]) -> Result<&mut V, ViewError> {
         use std::collections::btree_map::Entry;
 
-        let update = match self.updates.entry(short_key.clone()) {
+        let update = match self.updates.entry(short_key.to_vec()) {
             Entry::Vacant(e) if self.delete_storage_first => e.insert(Update::Set(V::default())),
             Entry::Vacant(e) => {
-                let key = self.context.base_index(&short_key);
-                let value = self.context.read_value(&key).await?.unwrap_or_default();
-                e.insert(Update::Set(value))
+                if contains_key(&self.deleted_prefixes, short_key) {
```

Comment:
This case could be shared above `if self.delete_storage_first || contains_key(&self.deleted_prefixes, short_key) `

Reply:
Yes, that is correct,